pricelist_price_base_custom/models/product_pricelist_item.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models, tools
+
+
+class PricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    base = fields.Selection(
+        selection_add=[("custom_value", "Custom Value")],
+        ondelete={"custom_value": "set default"},
+    )
+
+    def _compute_price(self, product, quantity, uom, date, currency=None):
+        """
+        Overridden method to add yet another way to calculate the price
+        depending on the custom_base_price from the context of recordset.
+        """
+        if (
+            "custom_base_price" in self.env.context
+            and self.compute_price == "formula"
+            and self.base == "custom_value"
+        ):
+            product_uom = product.uom_id
+            if product_uom != uom:
+                convert = lambda p: product_uom._compute_price(p, uom)  # noqa: E731
+            else:
+                convert = lambda p: p  # noqa: E731
+            base_price = self.env.context.get("custom_base_price")
```

Comment:
Why not simply to override the [_compute_base_price](https://github.com/odoo/odoo/blob/638b8255b7956dac868d4e35e4a101f3bc860a2f/addons/product/models/product_pricelist_item.py#L403) method?